### PR TITLE
Migrate SettingsController to kotlin + viewbinding

### PR DESCRIPTION
**Pull request by AndyScherzinger**
Signed-off-by: Andy Scherzinger <info@andy-scherzinger.de>

### CODE REVIEW

File: app/src/main/java/com/nextcloud/talk/controllers/SettingsController.kt [hunk 1]
```diff
@@ -0,0 +1,1065 @@
...
+                            binding.settingsProxyPortEdit.value = "1080"
+                        else -> {
+                        }
+                    }
+                    showProxySettings()
+                }
+            } catch (npe: NullPointerException) {
+                // view binding can be null
+                // since this is called asynchronously and UI might have been destroyed in the meantime
+                Log.i(ProfileController.TAG, "UI destroyed - view binding already gone")
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/controllers/SettingsController.kt:881]: Log.i(TAG, ...

File: app/src/main/java/com/nextcloud/talk/controllers/ProfileController.kt [hunk 2]
```diff
@@ -808,7 +808,7 @@ class ProfileController : NewBaseController(R.layout.controller_profile) {
     }
 
     companion object {
-        private const val TAG: String = "ProfileController"
```
Comment (mahibi) [inline app/src/main/java/com/nextcloud/talk/controllers/ProfileController.kt:811]: can be private (don't import to SettingsController)

**Review by mahibi — APPROVED** [review]

**Comment by mahibi** [comment]
:+1: :+1:

File: app/src/main/java/com/nextcloud/talk/controllers/SettingsController.kt (hunk 1 above)
Comment (AndyScherzinger) [reply app/src/main/java/com/nextcloud/talk/controllers/SettingsController.kt]: Fixed 👍

**Comment by AndyScherzinger** [comment]
review comments applied, waiting for CI now 👍 Thanks for all you reviews @mahibi @timkrueger 🙏

**Pull request merged**